string stream_name
bool on
---
bool success
